versions/releases.py

```diff
@@ -181,11 +181,16 @@ def get_release_notes_for_version_s3(version_pk):
     # Note we are using the non-beta slug since release notes for beta
     # versions are named without beta suffix.
     filename = version.non_beta_slug.replace("-", "_")
-    response = get_file_data(
-        get_s3_client(),
-        settings.STATIC_CONTENT_BUCKET_NAME,
-        f"release-notes/master/{filename}.adoc",
-    )
+    s3_client = get_s3_client()
```

Comment:
I'd think we should tweak this so we have a set of alternative strings/formats checked in a while loop, that way there's flexibility for adding new ones in the future, and we only have the call code once.

Reply:
I considered this, but

1. I wasn't really sure what other patterns might exist, making it a bit difficult to generalize, and
2. [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it#:~:text=%22You%20aren't%20gonna%20need,add%20functionality%20until%20deemed%20necessary.) - I figured we could add this once we have more than two known possible patterns

If you think we should still add this now, I'm open to doing so, though we will still need a code change to at a minimum add the third pattern if it crops up. LMK your thoughts.

Reply:
Personally I'd have it not repeat the code in both places, but it will work as is so I've approved the PR.

Reply:
Fair - I have a feeling we'll be revisiting this code, and I'll keep this in mind. I'm going to add a TODO comment but merge this as-is for now. Thanks Dave!